Source/Core/VideoBackends/D3D/TextureCache.cpp

```diff
@@ -179,17 +181,165 @@ void TextureCache::TCacheEntry::FromRenderTarget(u32 dstAddr, unsigned int dstFo
 
 		size_in_bytes = (u32)encoded_size;
 
-		TextureCache::MakeRangeDynamic(addr, (u32)encoded_size);
+		TextureCache::MakeRangeDynamic(dstAddr, (u32)encoded_size);
 
 		this->hash = hash;
 	}
 }
 
+const char palette_shader[] =
+R"HLSL(
+sampler samp0 : register(s0);
+Texture2DArray Tex0 : register(t0);
+Buffer<uint> Tex1 : register(t1);
+uniform float Multiply;
+
+uint Convert3To8(uint v)
+{
+	// Swizzle bits: 00000123 -> 12312312
+	return (v << 5) | (v << 2) | (v >> 1);
```

Comment:
There is nowhere you don't use these functions without an &7 (or however many bits)

Might as well build the masking into these functions.